linera-sdk/src/contract/runtime.rs

```diff
@@ -172,3 +192,69 @@ where
             .expect("Failed to deserialize `Response` type from cross-application call")
     }
 }
+
+/// A helper type that uses the builder pattern to configure how a message is sent, and then
+/// sends the message once it is dropped.
+pub struct MessageSender<Message>
+where
+    Message: Serialize,
+{
+    message: OutgoingMessage<Message>,
+}
+
+impl<Message> MessageSender<Message>
+where
+    Message: Serialize,
+{
+    /// Creates a new [`MessageSender`] instance to send the `message` to the `destination`.
+    pub fn new(destination: Destination, message: Message) -> Self {
```

Comment:
Should we make this `pub(crate)` to make it even harder to accidentally send a message in the wrong format?

Reply:
Definitely, thanks for catching this!